jina/hub/encoders/numeric/tsne.py

```diff
@@ -2,21 +2,21 @@
 __license__ = "Apache-2.0"
 
 import numpy as np
-from .. import BaseNumericEncoder
-from ...decorators import batching
 
+from jina.executors.decorators import batching
+from . import TransformEncoder
 
-class TSNEEncoder(BaseNumericEncoder):
+
+class TSNEEncoder(TransformEncoder):
```

Comment:
`TSNEEncoder` does not fit into `TransformEncoder` because it cannot run `transform` without `fit`